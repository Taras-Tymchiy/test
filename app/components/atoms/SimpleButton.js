// @flow
import * as React from 'react';
import {
    Platform,
    StyleSheet,
    Button,
    Text,
    View,
    TouchableOpacity
  } from 'react-native';
import { COLOR_IOS_BUTTON_TEXT, FONT_SIZE_M } from '../styles/constants';
import { textRegular, textPrimary } from '../styles/mixins';

export interface SimpleButtonProps {
    +title: string;
    +style?: any;
    +textStyle?: any;
    +onPress: () => void;
}

export default function SimpleButton({title, onPress, style, textStyle}: SimpleButtonProps) {
  return (
    <TouchableOpacity onPress={onPress} style={style}>
      <Text style={[styles.regular, textStyle]} >{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  regular: {
    ...textRegular,
    color: COLOR_IOS_BUTTON_TEXT
  }
});