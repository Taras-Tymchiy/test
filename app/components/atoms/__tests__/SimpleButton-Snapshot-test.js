import React from 'react';
import renderer from 'react-test-renderer';
import SimpleButton, { SimpleButtonProps } from '../SimpleButton';

const props: SimpleButtonProps = {
    title: 'test title',
    onPress: () => null
};

it('renders a SimpleButton using Snapshots', () => {
  expect(renderer.create(
    <SimpleButton {...props} />
  )).toMatchSnapshot();
});

it('renders a SimpleButton with custom styles using Snapshots', () => {
  const newProps = {
    ...props,
    style: {
      backgroundColor: '#FFF'
    }
  };
  expect(renderer.create(
    <SimpleButton {...newProps} />
  )).toMatchSnapshot();
});

it('renders a SimpleButton with custom text styles using Snapshots', () => {
  const newProps = {
    ...props,
    textStyle: {
      color: '#FFF'
    }
  };
  expect(renderer.create(
    <SimpleButton {...newProps} />
  )).toMatchSnapshot();
});