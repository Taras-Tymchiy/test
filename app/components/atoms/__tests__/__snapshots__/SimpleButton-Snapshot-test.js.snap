// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a SimpleButton using Snapshots 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#037aff",
          "fontSize": 16,
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    test title
  </Text>
</View>
`;

exports[`renders a SimpleButton with custom styles using Snapshots 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFF",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#037aff",
          "fontSize": 16,
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    test title
  </Text>
</View>
`;

exports[`renders a SimpleButton with custom text styles using Snapshots 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#037aff",
          "fontSize": 16,
          "fontWeight": "normal",
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
  >
    test title
  </Text>
</View>
`;
